homeassistant/components/google_assistant/helpers.py

```diff
@@ -608,12 +565,31 @@ async def sync_serialize(self, agent_user_id):
         if room := entity_config.get(CONF_ROOM_HINT):
             device["roomHint"] = room
         else:
-            area = await _get_area(self.hass, entity_entry, device_entry)
-            if area and area.name:
+            if entity_entry and entity_entry.area_id:
+                area_id = entity_entry.area_id
+            elif device_entry and device_entry.area_id:
+                area_id = device_entry.area_id
+            else:
+                area_id = None
+
+            if (
+                area_id
+                and (area := area_registry.async_get(self.hass).async_get_area(area_id))
+                and area.name
+            ):
```

Comment:
I hink the helper made sense here's. The amount of code in this function get hard to read.